pkg/certifier/datadog_malware/datadog_malware.go

```diff
@@ -0,0 +1,241 @@
+//
+// Copyright 2024 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadog_malware
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/guacsec/guac/pkg/assembler"
+	"github.com/guacsec/guac/pkg/assembler/clients/generated"
+	ingestor "github.com/guacsec/guac/pkg/assembler/clients/helpers"
+	"github.com/guacsec/guac/pkg/assembler/helpers"
+	"github.com/guacsec/guac/pkg/certifier"
+	"github.com/guacsec/guac/pkg/certifier/components/root_package"
+	"github.com/guacsec/guac/pkg/clients"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+	"github.com/guacsec/guac/pkg/version"
+	jsoniter "github.com/json-iterator/go"
+	"golang.org/x/time/rate"
+)
+
+var (
+	json              = jsoniter.ConfigCompatibleWithStandardLibrary
+	rateLimit         = 10000
+	rateLimitInterval = 30 * time.Second
+)
+
+const (
+	NPM_MANIFEST_URL        string = "https://raw.githubusercontent.com/DataDog/malicious-software-packages-dataset/main/samples/npm/manifest.json"
+	PYPI_MANIFEST_URL       string = "https://raw.githubusercontent.com/DataDog/malicious-software-packages-dataset/main/samples/pypi/manifest.json"
+	DatadogMalwareCertifier string = "datadog_malware_certifier"
+)
+
+var ErrDatadogMalwareComponentTypeMismatch error = errors.New("rootComponent type is not []*root_package.PackageNode")
+
+type MaliciousPackages map[string][]string
+
+type assemblerFuncType func([]assembler.IngestPredicates) (*ingestor.AssemblerIngestedIDs, error)
+
+type datadogMalwareCertifier struct {
+	httpClient    *http.Client
+	npmData       MaliciousPackages
+	pypiData      MaliciousPackages
+	assemblerFunc assemblerFuncType
+}
+
+// CertifierOption defines functional options for the certifier
+type CertifierOption func(*datadogMalwareCertifier)
+
+// WithHTTPClient allows overriding the default HTTP client
+func WithHTTPClient(client *http.Client) CertifierOption {
+	return func(d *datadogMalwareCertifier) {
+		d.httpClient = client
+	}
+}
+
+// NewDatadogMalwareCertifier initializes the Datadog Malicious Software Packages certifier
+func NewDatadogMalwareCertifier(ctx context.Context, assemblerFunc assemblerFuncType, opts ...CertifierOption) (certifier.Certifier, error) {
```

Comment:
Can you add a bit of documentation here on what the datadog malicious software packages are, for those that are not familiar.

In addition could you add some details on:
- The added predicates on the graph
- The recommended interval times (considering that the current certifier will generate a certifyBad each time indefinitely).
- Any caveats: see comment on periodic fetching of manifest.

Reply:
Sure thing, I'll add some documentation along these lines